ui/packages/lib/src/hooks/useDynamicScript.js

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+
+const urlCache = new Set();
```

Comment:
I guess this file is largely similar to [useDynamicScript.js](https://github.com/caraml-dev/turing/blob/main/ui/src/hooks/useDynamicScript.js). But the `urlCache` manipulation is new - what does it achieve?

Reply:
Both `useDynamicScript` and `useFederatedComponent` hooks were taken from the official [`module-federation/module-federation-examples`](https://github.com/module-federation/module-federation-examples) repository (the link to the impl included in the in-line comment two lines below your question). I'll be surprised to see the implementation of these hooks to be very different in the `caraml-dev/turing` repository :) 

As regards the `urlCache`, it allows to avoid inserting the new `<script />` element into the document tree, if it was already done before. So `remoteEntry.js` is downloaded only once for each of the remote applications, even if you have multiple different modules exposed by the same remote app.

Reply:
Oops, I assumed you had adapted this file from the Turing codebase (which took it from the same source earlier) and added some enhancements. Thanks for the explanation.